### DISCUSSION TITLE
Will 8/16-bit types be supported?

### DISCUSSION THREAD

**Discussion by hw762** [gpuweb/gpuweb#3412] [Q&A] [answered]
I remember earlier drafts of WGSL seemed to have reserved words of `u8`, `u16`, etc. Will these scalars be supported in a future version? I understand that GLSL only has 32-bit scalars, but it'd be convenient if I can bind an array of `u8` instead of doing bit manipulation manually.

**Comment by alan-baker** [comment, marked as answer]
These will likely come as extensions to the language after the initial release.
